src/Utilities/Functional.hpp

```diff
@@ -9,12 +9,67 @@
 
 #include "ErrorHandling/Assert.hpp"
 #include "ErrorHandling/StaticAssert.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Math.hpp"
 
 /*!
  * \ingroup UtilitiesGroup
  * \brief Higher order function objects similar to `std::plus`, etc.
+ *
+ * \details
+ * These chaining function objects can be used to represent highly general
+ * mathematical operations
+ * 1. as types, which can be passed around in template arguments, and
+ * 2. such that any time they can be evaluated at compile time, they will be.
+ *
+ * As an example usecase, consider a function templated on a generic analytic
+ * solution
+ * ```
+ * template <class Sol, class... Ts> sample(Ts... params) { ...
+ * ```
+ * In the corresponding namespace, we could use these functional objects to
+ * define useful distributions, or generate them in the user code.
+ * ```
+ * using Sinusoid = funcl::Multiplies<funcl::Identity,funcl::Sin<
```

Comment:
Add space after comma for readability :)

Also, can you turn this into a doxygen snippet?